hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestPartitionStatsIndex.scala

```diff
@@ -178,7 +177,7 @@ class TestPartitionStatsIndex extends PartitionStatsIndexTestBase {
       override def apply(): Boolean = {
         try {
           doWriteAndValidateDataAndPartitionStats(hudiOpts,
-            operation = DataSourceWriteOptions.UPSERT_OPERATION_OPT_VAL,
+            operation = DataSourceWriteOptions.BULK_INSERT_OPERATION_OPT_VAL,
```

Comment:
Could we add validation on the preconditions of the end state before validation, i.e., in this case there must be two commits with overlapping commit time span (based on the instant time and completion time)?

I think it might also be valuable to add the case where there are concurrent writers but the second writer has a conflict (add the validation on this precondition); and the partition stats index should not be affected.

Reply:
Done.